mono/metadata/mono-config.c

```diff
@@ -634,6 +637,7 @@ mono_config_for_assembly_internal (MonoImage *assembly)
 	mono_config_parse_file_with_context (&state, cfg_name);
 	g_free (cfg_name);
 
+#ifndef TARGET_ANDROID
```

Comment:
Would prefer if you added a minimal config instead of basing this on the target platform:
1. Add support for `--enable-minimal=cfgdir_config` to [configure.ac](https://github.com/mono/mono/blob/174f2f6a70084eced80d4c0284b0a9eb2cf25432/configure.ac#L1751) to define `DISABLE_CFGDIR_CONFIG` (or some better name)
2. Use that for the ifdef here.
3. Pass `--enable-minimal=cfgdir_config` in [android.mk for SDK builds](https://github.com/mono/mono/blob/174f2f6a70084eced80d4c0284b0a9eb2cf25432/sdks/builds/android.mk#L144) (there are multiple templates that will need to be updated)

Reply:
@lambdageek I only found one usage of `--enable-minimal` for Android. Are you saying I should add this for iOS and wasm, too?

Reply:
> I only found one usage of --enable-minimal for Android.

Sorry, my mistake - I thought there were a few uses of it in there.  Just Android is alright.